go.mod

```diff
@@ -19,15 +19,16 @@ require (
 	github.com/pmezard/go-difflib v1.0.0
 	github.com/prashantv/gostub v1.1.0
 	github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring v0.51.2
-	github.com/prometheus/client_golang v1.14.0
+	github.com/prometheus/client_golang v1.20.5
```

Comment:
let's not change client golang in the same time.

Reply:
It seems to be the cause of UT failure

Reply:
Likely the generate code is also out of date. These versions are over 2 years old. I think I have run into this before, it's an easy fix.

Reply:
Found the issue, the updated `stretchr/testify` required a couple tests to be fixed.